### PR TITLE
updated tutorial to v6

### PR DESCRIPTION
**Pull request by dzirtusss**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/506)

### CODE REVIEW

**Comment by mamute** [comment]
Just followed the updated tutorial:

**Comment by mamute** [comment]
Hey guys, just tried the updated instructions, very neat!

Here is repo: https://github.com/mamute/test-react-on-rails
The heroku app: https://test-react-on-rails.herokuapp.com/
And a GIF with the generated code + the magic happening: 

everything went fine except by this lines:
```
bin/rake db:migrate
bin/rake db:setup
```

Running in that order it throws: `ActiveRecord::NoDatabaseError: FATAL:  database "test-react-on-rails_development" does not exist`

**Comment by justin808** [comment]
@mamute Can you attach the gif directly to the PR?

**Comment by dzirtusss** [comment]
As I understand compared with previous versions, hot mode is not working anymore in generated default hello app (it needs Procfile.hot + correction in package.json scripts + etc.) which are not generated by default generator. So I removed that previous abstract from tutorial or ????

Plus updated Rails gem in setup (forgot) and custom IP/PORT. Now it can run from mostly blank linux.

will check Heroku part for errors.

Any good idea where to host screenshot images for tutorial.md?

**Comment by mamute** [comment]
Uploaded `https://cloud.githubusercontent.com/assets/371302/17368567/111cc722-596b-11e6-9b72-ac5967a60e42.gif`: gif

**Comment by justin808** [comment]
@dzirtusss you can just drag and drop screen shots right on this PR for hosting.

**Comment by justin808** [comment]
@mamute @dzirtusss Is this ready for merge?

**Comment by dzirtusss** [comment]
Not yet, need to check Heroku and add more screenshots

File: docs/tutorial.md [hunk 1]
```diff
@@ -14,135 +13,86 @@ By the time you read this, the latest may have changed. Be sure to check the ver
 * https://rubygems.org/gems/react_on_rails
 * https://www.npmjs.com/package/react-on-rails
 
-Trying out v4 is super easy, so long as you have the basic prerequisites. This includes the basics for Rails 4.x and node version 5+. I recommend `rvm` and `nvm` to install Ruby and Node.
+##Seting up environment
```
Comment (AlexKVal) [inline docs/tutorial.md]: I believe this line is meant to be:
`## Setting up the environment`

File: docs/tutorial.md [hunk 2]
```diff
@@ -14,135 +13,86 @@ By the time you read this, the latest may have changed. Be sure to check the ver
...
-cd <directory where you want to create your new Rails app>
-# any name you like
-rails new test-react-on-rails-3
-cd test-react-on-rails-3
-# git-extras command to make a new git repo and commit everything
-git setup
-vim Gemfile
-```
+nvm install 5.12            # download and install latest stable Node
+nvm alias default 5.12      # make 5.12 default version
```
Comment (AlexKVal) [inline docs/tutorial.md]: `nvm` allows to install the latest ( in this case 5 major) version like that:

```
> nvm install v5
> nvm alias default v5
```

File: docs/tutorial.md [hunk 3]
```diff
@@ -14,135 +13,86 @@ By the time you read this, the latest may have changed. Be sure to check the ver
...
-Add this line to your Gemfile:
+rvm install 2.3.1           # download and install latest stable Ruby
+rvm list                    # check
 
-```
-gem 'react_on_rails'
+gem install rails           # download and install latest stable Rails
 ```
 
+Than we need to do normal blank Rails app as following:
```
Comment (AlexKVal) [inline docs/tutorial.md]: `Thank` => `Then`
`to do` => `to make` / `to create`
probably smth like: `to create a fresh Rails application`

File: docs/tutorial.md [hunk 4]
```diff
@@ -14,135 +13,86 @@ By the time you read this, the latest may have changed. Be sure to check the ver
...
-npm run express-server
+gem 'react_on_rails', '~>6'         # use latest gem version > 6
 ```
 
-<img src="http://forum.shakacode.com/uploads/default/original/1X/5bd396a7f9c0764929b693fb79eb6685ec6f62cf.png" width="470" height="499">
-
-Visit http://localhost:4000 and see your React On Rails app running using the Webpack Dev server.
-
-With this setup, you can make changes to your JS or CSS and the browser will hot reload the changes (no page refresh required).
+put everything under git repository (or rails generate will not work properly)
```
Comment (AlexKVal) [inline docs/tutorial.md]: I would quote `rails generate` like that:

```
... (or `rails generate` will not work properly)
```

File: docs/tutorial.md [hunk 5]
```diff
@@ -14,135 +13,86 @@ By the time you read this, the latest may have changed. Be sure to check the ver
...
 
-I'm going to add this line to client/app/bundles/HelloWorld/HelloWorldWidget.jsx:
-
-```html
-<h1>Welcome to React On Rails!</h1>
+```
+# git command to make a new git repo and commit everything
+git init
+git add -A
+git commit -m "First commit"
```
Comment (AlexKVal) [inline docs/tutorial.md]: usually `Initial commit` is more popular, but it doesn't really matter :)

File: docs/tutorial.md [hunk 6]
```diff
@@ -14,135 +13,86 @@ By the time you read this, the latest may have changed. Be sure to check the ver
...
-
-With this setup, you can make changes to your JS or CSS and the browser will hot reload the changes (no page refresh required).
+put everything under git repository (or rails generate will not work properly)
 
-I'm going to add this line to client/app/bundles/HelloWorld/HelloWorldWidget.jsx:
-
-```html
-<h1>Welcome to React On Rails!</h1>
+```
+# git command to make a new git repo and commit everything
```
Comment (AlexKVal) [inline docs/tutorial.md]:
```
# Here are git commands ...
```

File: docs/tutorial.md [hunk 7]
```diff
@@ -14,135 +13,86 @@ By the time you read this, the latest may have changed. Be sure to check the ver
...
 ```
 
-    bin/rake lint
-
-The edit `application.scss`. Delete the comment at the top.
-
-    bin/rake lint
-
-You should see 
+and than run server with
```
Comment (AlexKVal) [inline docs/tutorial.md]: `than` => `then`

File: docs/tutorial.md [hunk 8]
```diff
@@ -14,135 +13,86 @@ By the time you read this, the latest may have changed. Be sure to check the ver
...
-> jscs --verbose . "."
+In case you are running some custom setup with different IP or PORT you should also edit Procfile.dev. For example to be able to run on free Cloud9 IDE we are putting IP 0.0.0.0 and PORT 8080
 
-Completed running all JavaScript Linters
-Completed all linting
+``` Procfile.dev
+web: rails s -p 8080 -b 0.0.0.0
 ```
 
+Than visit  https://your-shared-addr.c9users.io:8080/hello_world
```
Comment (AlexKVal) [inline docs/tutorial.md]: `Than` => `Then`

**Comment by justin808** [comment]
@AlexKVal, thanks for the feedback! Any other suggestions?

**Comment by justin808** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[docs/tutorial.md, line 16 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/506#-KOBec9K-rYJVv0BNQZ7-r2-16:-KOWCD1Xd8KsUjd8lTPE:b-7hy70p) ([raw file](https://github.com/shakacode/react_on_rails/blob/b02ebc2cd2e49a3a9844e9fd51f4faab2e8b3446/docs/tutorial.md#L16)):*
<details><summary><i>Previously, AlexKVal (Alexander Shemetovsky) wrote…</i></summary>
> I believe this line is meant to be:
> 
> `## Setting up the environment`
</details>
:thumbsup: 

---

*[docs/tutorial.md, line 22 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/506#-KOBec9K-rYJVv0BNQZ7-r2-22:-KOWCRWwp-kVkkumGLAE:b-fwc35f) ([raw file](https://github.com/shakacode/react_on_rails/blob/b02ebc2cd2e49a3a9844e9fd51f4faab2e8b3446/docs/tutorial.md#L22)):*
<details><summary><i>Previously, AlexKVal (Alexander Shemetovsky) wrote…</i></summary>
> `nvm` allows to install the latest ( in this case 5 major) version like that:
> 
> ```
> 
> > nvm install v5
> 
> > nvm alias default v5
> 
> ```
> 
> 
</details>
6.3.1 is now current: https://github.com/creationix/nvm#long-term-support

```
nvm install node
```

Basically, we should say we intend to support the current version, and older versions may or may not be supported.

---

*[docs/tutorial.md, line 31 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/506#-KOBec9K-rYJVv0BNQZ7-r2-31:-KOWDGLXUjV5IbzCOUZE:b-jd3vnx) ([raw file](https://github.com/shakacode/react_on_rails/blob/b02ebc2cd2e49a3a9844e9fd51f4faab2e8b3446/docs/tutorial.md#L31)):*
<details><summary><i>Previously, AlexKVal (Alexander Shemetovsky) wrote…</i></summary>
> `Thank` => `Then`
> 
> `to do` => `to make` / `to create`
> 
> probably smth like: `to create a fresh Rails application`
> 
> 
</details>
:thumbsup: 

---

*[docs/tutorial.md, line 47 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/506#-KOBec9K-rYJVv0BNQZ7-r2-47:-KOWDLTp_aI_AtipuhaU:b4c9lyr) ([raw file](https://github.com/shakacode/react_on_rails/blob/b02ebc2cd2e49a3a9844e9fd51f4faab2e8b3446/docs/tutorial.md#L47)):*
<details><summary><i>Previously, AlexKVal (Alexander Shemetovsky) wrote…</i></summary>
> I would quote `rails generate` like that:
> 
> ```
> 
> ... (or `rails generate` will not work properly)
> 
> ```
</details>
:thumbsup: 

---

*[docs/tutorial.md, line 50 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/506#-KOBec9K-rYJVv0BNQZ7-r2-50:-KOWDNv1DtuyUY3uHTgE:b4c9lyr) ([raw file](https://github.com/shakacode/react_on_rails/blob/b02ebc2cd2e49a3a9844e9fd51f4faab2e8b3446/docs/tutorial.md#L50)):*
<details><summary><i>Previously, AlexKVal (Alexander Shemetovsky) wrote…</i></summary>
> ```
> 
> # Here are git commands ...
> 
> ```
</details>
:thumbsup: 

---

*[docs/tutorial.md, line 53 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/506#-KOBec9K-rYJVv0BNQZ7-r2-53:-KOWDQeT3Bh5DAw090KZ:b-8fx2u7) ([raw file](https://github.com/shakacode/react_on_rails/blob/b02ebc2cd2e49a3a9844e9fd51f4faab2e8b3446/docs/tutorial.md#L53)):*
<details><summary><i>Previously, AlexKVal (Alexander Shemetovsky) wrote…</i></summary>
> usually `Initial commit` is more popular, but it doesn't really matter :)
</details>
:+1: 

---

*[docs/tutorial.md, line 64 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/506#-KOBec9K-rYJVv0BNQZ7-r2-64:-KOWDToEnBTd5lhVQ_th:b-8fx2u7) ([raw file](https://github.com/shakacode/react_on_rails/blob/b02ebc2cd2e49a3a9844e9fd51f4faab2e8b3446/docs/tutorial.md#L64)):*
<details><summary><i>Previously, AlexKVal (Alexander Shemetovsky) wrote…</i></summary>
> `than` => `then`
</details>
:+1: 

---

*[docs/tutorial.md, line 80 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/506#-KOBec9K-rYJVv0BNQZ7-r2-80:-KOWDVh_nHSYGntyJ7ob:b-8fx2u7) ([raw file](https://github.com/shakacode/react_on_rails/blob/b02ebc2cd2e49a3a9844e9fd51f4faab2e8b3446/docs/tutorial.md#L80)):*
<details><summary><i>Previously, AlexKVal (Alexander Shemetovsky) wrote…</i></summary>
> `Than` => `Then`
</details>
:+1: 

---

*[docs/tutorial.md, line 134 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/506#-KOWEIanfP-aiKYwseni:-KOWEIanfP-aiKYwsenj:b8z9r7t) ([raw file](https://github.com/shakacode/react_on_rails/blob/63436ab2aef2f6f4c70ba9d329f39660264b9c2e/docs/tutorial.md#L134)):*
> ```Markdown
> * `client/node_modules`
> 
> 
> ```

we want to get some images in here.

Let's host them on github

drag the images into this PR.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/506)*

**Comment by AlexKVal** [comment]
@justin808 
> @AlexKVal, thanks for the feedback! Any other suggestions?

Particularly on this PR I have no other suggestions as for now 😄

**Comment by dzirtusss** [comment]
Looks like final version now. Updated all corrections, updated Heroku section, added screenshots.
Will study `than` & `then` :confused:

**Comment by justin808** [comment]
:lgtm_strong: 

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/506#-:-KOdVU4l83uW2CbjEbZG:b-uwkowd)*

**Comment by justin808** [comment]
Big thanks @dzirtusss!

**Pull request merged**